### PR TITLE
Add a `realpath` wrapper function

### PR DESCRIPTION
**Pull request by mckern**
If `realpath` isn't found in $PATH, then use Python to fake it.

### CODE REVIEW

File: functions.sh
Comment (doofus-gluteus) [inline functions.sh]: Braces go on the next line!

File: functions.sh
Comment (doofus-gluteus) [inline functions.sh]: No, `then` goes on the next line!

File: functions.sh
Comment (doofus-gluteus) [inline functions.sh]: ☝️ braces! newline! were you raised in a barn?

File: functions.sh
Comment (diligent-amicus) [inline functions.sh]: Greg's Bash Wiki [recommends](https://mywiki.wooledge.org/BashPitfalls#function_foo.28.29) that the `function` keyword not be combined with parenthesis. Please use either:

```suggestion
function realpath {
```

or 

```suggestion
realpath() {
```

**Review by ghost — DISMISSED** [review]
If you want me to "act professional," I can tell you that I'm not interested. I'm sitting in my home office wearign a bathrobe. The same way I'm not going to start wearing ties, I'm also not going to buy into the fake politeness, the lying, the office politics and backstabbing, the passive aggressiveness, and the buzzwords. Because THAT is what "acting professionally" results in: people resort to all kinds of really nasty things because they are forced to act out their normal urges in unnatural ways.

There's **no way** I'll let this be merged until those braces and `then`s go on the next line.

**Review by ghost — DISMISSED** [review]
Please use either the `function <name> {` format, or `<name>() {` format, but not both. Either kind will be acceptable, but [Greg's Bash Wiki](https://mywiki.wooledge.org/BashPitfalls#function_foo.28.29) recommends that both styles not be mixed together. I don't think it's worth blocking this on and I'm happy to push up a commit addressing this to get this moving again.

**Review by ghost — APPROVED** [review]
raw

File: functions.sh
Comment (ghost) [inline functions.sh:345]: I don't want to carry a Python requirement. Just do this:

```bash
realpath()
{
    if [[ -z "${1}" ]]
    then
        return 1
    fi

    if ! type -P realpath
        echo "$(cd $(dirname $1); pwd)/$(basename ${1})"
    else
        command realpath "${1}"
    fi
}
```

Works great, no Python! You make things too complicated.

```
doofus@DONKEYBURG refactored-octo-doodle $ realpath $PWD
/Users/ryan/Documents/Repositories/refactored-octo-doodle
```

Comment (ghost) [reply functions.sh:345]: Did you test this? That's not going to work right in many cases, but Python (which is already bundled onto every system we operate) will:

```bash
dilly@lappy refactored-octo-doodle $ ( set -x; command realpath /usr/local/bin/jq )
+ command realpath /usr/local/bin/jq
+ realpath /usr/local/bin/jq
/usr/local/Cellar/jq/1.6/bin/jq    # Correct


dilly@lappy refactored-octo-doodle $ ( set -x; realestpath /usr/local/bin/jq )
+ realestpath /usr/local/bin/jq
+++ dirname /usr/local/bin/jq
++ cd /usr/local/bin
++ pwd
++ basename /usr/local/bin/jq
+ echo /usr/local/bin/jq
/usr/local/bin/jq    # Incorrect


dilly@lappy refactored-octo-doodle $ python -c "import os; print(os.path.realpath('/usr/local/bin/jq'))"
/usr/local/Cellar/jq/1.6/bin/jq    # Correct
```

Here's a [good thread](https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac) about why "simple" problems often have complicated solutions. You should review it for context on this problem in particular.

**Comment by ghost** [comment]
No one needs this.

**Comment by mckern** [comment]
Why was this closed? I was working on the changes you suggested, and you closed this out from underneath me. Then you subtweeted my work on main -- that's really unprofessional.

**Pull request closed without merging**
